subxt/src/client/offline_client.rs

```diff
@@ -21,6 +21,8 @@ pub trait OfflineClientT<T: Config>: Clone + Send + Sync + 'static {
     fn genesis_hash(&self) -> T::Hash;
     /// Return the provided [`RuntimeVersion`].
     fn runtime_version(&self) -> RuntimeVersion;
+    /// Return the Client Metadata.
+    fn client_metadata(&self) -> ClientState<T>;
```

Comment:
I don't get why we need the client_metadata here since it contains metadata, genesis_hash and runtime_version? 
To it's redundant to the old APIs?

Reply:
It is a bit redundant, I agree. But we need some bundled form of the client data to use it in the creation of signed extensions. They we previously generic over the type of client, but now the signed extensions live in `subxt-core` but the client traits should stay in `subxt`. I did not want to remove the old methods either because if you just want to get e.g. the `runtime_version` there is no need to clone the `Arc` the metadata is wrapped up in.